Frameworks/include/MockClass.h

```diff
@@ -0,0 +1,824 @@
+
+// Mock a seven-arg non-const method in low level C++
+#define MOCK_STDCALL_METHOD_7(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 7, false, __stdcall)
+
+// Mock an eight-arg non-const method in low level C++
+#define MOCK_STDCALL_METHOD_8(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 8, false, __stdcall)
+
+// Mock a nine-arg non-const method in low level C++
+#define MOCK_STDCALL_METHOD_9(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 9, false, __stdcall)
+
+// Mock a ten-arg non-const method in low level C++
+#define MOCK_STDCALL_METHOD_10(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 10, false, __stdcall)
+
+///////////////////////////////////////////////////////////////////
+// Macros for mocking const __stdcall methods in low level C++
+///////////////////////////////////////////////////////////////////
+
+// Mock a zero-arg const method in low level C++
+#define MOCK_CONST_STDCALL_METHOD_0(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 0, true, __stdcall)
+
+// Mock a one-arg const method in low level C++
+#define MOCK_CONST_STDCALL_METHOD_1(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 1, true, __stdcall)
+
+// Mock a two-arg const method in low level C++
+#define MOCK_CONST_STDCALL_METHOD_2(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 2, true, __stdcall)
+
+// Mock a three-arg const method in low level C++
+#define MOCK_CONST_STDCALL_METHOD_3(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 3, true, __stdcall)
+
+// Mock a four-arg const method in low level C++
+#define MOCK_CONST_STDCALL_METHOD_4(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 4, true, __stdcall)
+
+// Mock a five-arg const method in low level C++
+#define MOCK_CONST_STDCALL_METHOD_5(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 5, true, __stdcall)
+
+// Mock a six-arg const method in low level C++
+#define MOCK_CONST_STDCALL_METHOD_6(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 6, true, __stdcall)
+
+// Mock a seven-arg const method in low level C++
+#define MOCK_CONST_STDCALL_METHOD_7(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 7, true, __stdcall)
+
+// Mock an eight-arg const method in low level C++
+#define MOCK_CONST_STDCALL_METHOD_8(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 8, true, __stdcall)
+
+// Mock a nine-arg const method in low level C++
+#define MOCK_CONST_STDCALL_METHOD_9(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 9, true, __stdcall)
+
+// Mock a ten-arg const method in low level C++
+#define MOCK_CONST_STDCALL_METHOD_10(METHOD_NAME) BUILD_MOCK_METHOD_IMPL(METHOD_NAME, 10, true, __stdcall)
+
+/*/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+C++/CX usage is slightly different than it is for low level C++:
+
+1. Due to an issue in the compiler, you must first call CX_MOCK_INTERFACE before any of the CX_MOCK* macros will compile.
+
+CX_MOCK_INTERFACE(AnotherNamespace::ITestInterface);
```

Comment:
Might be worth looking at CX_MOCK_INTERFACE to see if we could have borrowed code / concepts for our MOCK_CLASS macro.